src/base/client.zig

```diff
@@ -232,15 +235,20 @@ pub fn BaseClient(comptime Reader: type, comptime Writer: type) type {
             }
         }
 
+        pub fn closeReader(self: *Self) void {
```

Comment:
Extracted this since the parser can terminate early — e.g. not consuming trailing letters once all the data has been located. It's simpler and faster to `defer closeReader()` than manually consuming the reader.

Caveat: this will not consume any remaining *chunks*, instead it only force the client back to non-reader mode so it doesn't crash when followed up with a manual `client.next()`. Should this be changed?